flink-python/pyflink/datastream/window.py

```diff
@@ -517,3 +538,721 @@ def __init__(self,
+                           window: CountWindow,
+                           ctx: Trigger.TriggerContext) -> TriggerResult:
+        # No action is taken on the window.
+        return TriggerResult.CONTINUE
+
+    def on_event_time(self,
+                      time: int,
+                      window: CountWindow,
+                      ctx: Trigger.TriggerContext) -> TriggerResult:
+        # No action is taken on the window.
+        return TriggerResult.CONTINUE
+
+    def on_merge(self, window: CountWindow, ctx: Trigger.OnMergeContext) -> None:
+        ctx.merge_partitioned_state(self._count_state_descriptor)
+
+    def can_merge(self) -> bool:
+        return True
+
+    def clear(self, window: CountWindow, ctx: Trigger.TriggerContext) -> None:
+        state = ctx.get_partitioned_state(self._count_state_descriptor)
+        state.clear()
+
+
+class TumblingWindowAssigner(WindowAssigner[T, TimeWindow]):
+    """
+    A WindowAssigner that windows elements into windows based on the current system time of the
+    machine the operation is running on. Windows cannot overlap.
+    For example, in order to window into windows of 1 minute, every 10 seconds:
+    ::
+            >>> data_stream.key_by(lambda x: x[0], key_type=Types.STRING()) \
+            >>> .window(TumblingWindowAssigner(Time.minutes(1), Time.seconds(10), False))
+
+    A WindowAssigner that windows elements into windows based on the timestamp of the elements.
+    Windows cannot overlap.
+    For example, in order to window into windows of 1 minute:
+     ::
+            >>> data_stream.assign_timestamps_and_watermarks(watermark_strategy) \
+            >>> .key_by(lambda x: x[0], key_type=Types.STRING()) \
+            >>> .window(TumblingWindowAssigner(60000, 0, True))
+    """
+
+    def __init__(self, size: int, offset: int, is_event_time: bool):
+        if abs(offset) >= size:
+            raise Exception("TumblingWindowAssigner parameters must satisfy abs(offset) < size")
+
+        self._size = size
+        self._offset = offset
+        self._is_event_time = is_event_time
+
+    def assign_windows(self,
+                       element: T,
+                       timestamp: int,
+                       context: WindowAssigner.WindowAssignerContext) -> Collection[TimeWindow]:
+        if self._is_event_time is False:
+            current_processing_time = context.get_current_processing_time()
+            start = TimeWindow.get_window_start_with_offset(current_processing_time, self._offset,
+                                                            self._size)
+            return [TimeWindow(start, start + self._size)]
+        else:
+            if timestamp > MIN_LONG_VALUE:
+                start = TimeWindow.get_window_start_with_offset(timestamp, self._offset, self._size)
+                return [TimeWindow(start, start + self._size)]
+            else:
+                raise Exception("Record has MIN_LONG_VALUE timestamp (= no timestamp marker). "
```

Comment:
use the value of MIN_LONG_VALUE

Reply:
maybe use `jvm.java.lang.Long.MIN_VALUE` instead?